EU4toV2/Source/EU4World/Provinces/Provinces.h

```diff
@@ -1,36 +1,36 @@
 #ifndef PROVINCES_H
 #define PROVINCES_H
-
+#include "Configuration.h"
 #include "EU4Province.h"
 #include "Parser.h"
-#include "../../Configuration.h"
 #include <map>
 
-namespace mappers {
-	class SuperGroupMapper;
+namespace mappers
+{
+class SuperGroupMapper;
 }
 
 namespace EU4
 {
-	class Regions;
+class Regions;
+class Provinces: commonItems::parser
+{
+  public:
+	explicit Provinces(std::istream& theStream);
 
-	class Provinces: commonItems::parser
-	{
-	public:
-		explicit Provinces(std::istream& theStream);
-		
-		[[nodiscard]] auto geTotalProvinceWeights() const { return totalProvinceWeights; }
-		[[nodiscard]] const auto& getAllProvinces() const { return provinces; }
-		std::shared_ptr<Province> getProvince(int provinceNumber);
-		
-		void determineTotalProvinceWeights(const Configuration& configuration);
+	[[nodiscard]] auto geTotalProvinceWeights() const { return totalProvinceWeights; }
```

Comment:
Add tests

Reply:
Seriously now:
![image](https://user-images.githubusercontent.com/2392697/99079174-dbb53800-25bf-11eb-8321-d8c6aa7aff77.png)
I don't even comprehend this code. To test it I'd have to understand it.

I'll see if I can break it down into something human-readible.

Reply:
Oh yes, that. It's one of those bits I meant to just keep refactoring bit by bit. Skip for now, but maybe open an issue to come back to it.

I suspect some bits are trying to replicate EU4 logic for things we can read. Maybe with some modifiers. Any such modifiers should be moved from code to a configurable.